### ISSUE TITLE
Automatically include paths leading up to an included path.

### ISSUE THREAD

**Issue by imsodin** [syncthing/syncthing#4364]
Scenario: Include a specific path and exclude everything else.

Current: You need to manually include all path components:
```
!/foo/bar/baz
!/foo/bar
!/foo
*
```

Proposal: Automatically expand an exclude line to also exclude all the directories leading up to the excluded path. So the same result as above can be achieved by
```
!/foo/bar/baz
*
```

**Comment by AudriusButkevicius** [comment]
I wanted to do that, but what happens with:
```
!foo/**/bar
```

**Comment by imsodin** [comment]
This was bound to happen when I label something easy xD
Actually even easier examples like any unanchored path is a problem: For `!foo/bar/baz` we would unignore any paths matching `foo` and `foo/bar` regardless whether `bar` and `baz` follow.

**Comment by calmh** [comment]
I don't think `!foo/**/bar` works today in any reasonable fashion?

**Comment by AudriusButkevicius** [comment]
Maybe we just axe double star, the rest we could handle.

**Comment by imsodin** [comment]
How do you handle the problem I brought up above about relative paths without asterisks?

**Comment by calmh** [comment]
How would you use the unanchored relative path unignore today (other than via the side effect that it also works as the anchored equivalent)?

**Comment by AudriusButkevicius** [comment]
Perhaps we can this do this for non-asterix cases, or when asterix is at the end (or we have a single asterix, not a double).

**Comment by imsodin** [comment]
I can't think of a real scenario, but lets get uncreative and say I am in a habit of creating some utility dir "utils" and the utils create some kind of "utils/cache" dir that I do not care about at all. I can ignore all of them by using `utils/cache`. Well now I can't think of any reason I want to do `!utils/cache/something`, but say I do, then every the automatically generated `!utils/cache` conflicts with `utils/cache`.

**Comment by AudriusButkevicius** [comment]
Also, I am not sure I understand what you guys are talking about

**Comment by imsodin** [comment]
Fixed but missed in #4811
